### DISCUSSION TITLE
After package updates, My _targets.R doesn't run

### DISCUSSION THREAD

**Discussion by kkami1115** [ropensci/targets#654] [Help]
I updated my R library by doing update.packages() this morning, but after that tar_glimpse() and tar_make() started giving me the following error.
> Error : The target script file (default: _targets.R) must end with a list of tar_target() objects (recommended) or a tar_pipeline() object (deprecated). Each element of the target list must be a target object or nested list of target objects.
> Error: callr subprocess failed: The target script file (default: _targets.R) must end with a list of tar_target() objects (recommended) or a tar_pipeline() object (deprecated). Each element of the target list must be a target object or nested list of target objects.
> Visit https://books.ropensci.org/targets/debugging.html for debugging advice.

Of course, the script worked fine yesterday.
I don't know what I need to check next, so please tell me how I can check the circumstances.

Thanks in advance, 

>  sessionInfo()
R version 4.1.1 (2021-08-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)
Matrix products: default
locale:
[1] LC_COLLATE=Japanese_Japan.932  LC_CTYPE=Japanese_Japan.932    LC_MONETARY=Japanese_Japan.932 LC_NUMERIC=C                  
[5] LC_TIME=Japanese_Japan.932

**Comment by wlandau** [comment]
Not much I can add apart from the error message, unfortunately. Do you have a small example that reproduces the problem?

A target list looks something like `list(tar_target(name, command()), ...)`, and your `_targets.R` should have a target list at the very bottom.

**Comment by kkami1115** [comment]
After check I found my mistake.
I had mistakenly written an empty `tar_map()` in `functions.R` loaded by `source()`, instead of `_targets.R`.
It seems that if there is an empty `tar_map()`, `tar_glimpse()` or `tar_make()` will output the error shown above.
The package update had nothing to do with it. Sorry.

I'd like to ask you to deal with the `tar_map()` error, and I think it would be nice to have a clearer separation between `functions.R` and `_targets.R` (or a target list).
This would make it easier to handle errors.

**Reply by wlandau** [reply]
Just pushed a couple commits to allow empty input to `tar_map()` and improve the error message when checking for target lists.

**Reply by kkami1115** [reply]
Thanks! I'm working on pipelining my own script with {targets} and this is very impressive. Keep up the good work.
